pkg/controller/kieapp/defaults/merge.go

```diff
@@ -251,10 +249,19 @@ func mergeImageChangeParams(baseline appsv1.DeploymentTriggerImageChangeParams,
 	return baseline, nil
 }
 
+// findDeploymentTriggerPolicy Find a deploymentTrigger by Type. In case type == ImageChange
+// the match will be returned if both are not empty
 func findDeploymentTriggerPolicy(object appsv1.DeploymentTriggerPolicy, slice []appsv1.DeploymentTriggerPolicy) (int, appsv1.DeploymentTriggerPolicy) {
+	emptyImageChangeParams := &appsv1.DeploymentTriggerImageChangeParams{}
 	for index, candidate := range slice {
 		if candidate.Type == object.Type {
-			return index, candidate
+			if object.Type == appsv1.DeploymentTriggerOnImageChange {
+				if !reflect.DeepEqual(object.ImageChangeParams, emptyImageChangeParams) && !reflect.DeepEqual(candidate.ImageChangeParams, emptyImageChangeParams) {
```

Comment:
I believe a simple "!=" should suffice here

Reply:
No, it is not enough. It is failing to resolve as equal the case where the baseline has an empty ImageChangeParams and the override overrides the the image name.
Actually I preferred to use the go-cmp library instead of the reflect package.

Reply:
Can you please create a unit test method for this? I want to try locally changing it to "!=" and see why it doesn't work, cause it should.